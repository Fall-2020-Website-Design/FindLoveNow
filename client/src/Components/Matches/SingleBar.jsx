import styled from 'styled-components';

const SingleBar = styled.button`
    height: 7px;
    width: ${props => props.size};
    margin-left: 5px;
    background: ${props => props.color};
`;

export default SingleBar;