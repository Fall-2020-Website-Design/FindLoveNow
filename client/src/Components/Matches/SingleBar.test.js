import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SingleBar  from './SingleBar';


Enzyme.configure({ adapter: new Adapter() });

describe('SingleBar', () => {
    const params = {
        width: "10%;",
        color: "#FF0000"
    };

	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<SingleBar width={params.width} color={params.color} />);
    });
    
    test('renders', () => {
		const component = renderer.create(
			<MemoryRouter>
				<SingleBar />
			</MemoryRouter>
		).toJSON();
		
		expect(component).toMatchSnapshot();
    });
    

    test('check props', () => {
        expect(wrapper.prop('width')).toBe(params.width);
        expect(wrapper.prop('color')).toBe(params.color);
    });
    

});