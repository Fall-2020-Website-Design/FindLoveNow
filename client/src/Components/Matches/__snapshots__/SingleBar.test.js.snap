// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleBar renders 1`] = `
<button
  className="sc-bdfBwQ fFLQgX"
/>
`;
